File: LeetCode/0446/main.go
package main

import "fmt"

func numberOfArithmeticSlices(nums []int) int {
	n := len(nums)
	dp := make([]map[int]int, n)
	ans := 0
	for i := 1; i < n; i++ {
		dp[i] = make(map[int]int)
		for j := 0; j < i; j++ {
			d := nums[i] - nums[j]
			dp[i][d] += dp[j][d] + 1
			ans += dp[j][d]
		}
	}
	return ans
}

func main() {
	fmt.Println(
		numberOfArithmeticSlices(
			// []int{2, 4, 6, 8, 10},
			[]int{7, 7, 7, 7, 7},
		),
	)
}

File: LeetCode/0446/main.c
#include<stdio.h>
#include<stdlib.h>
#include<uthash.h>
#include<src/uthash.h>

typedef struct my_struct {
    int id;                    /* key */
    char name[10];
    UT_hash_handle hh;         /* makes this structure hashable */
}MyStruct;

int numberOfArithmeticSlices(int* nums, int numsSize){
    MyStruct *infos = NULL;
    
    
}

int main(int argc, char const *argv[])
{
    int num[1050];
    
    printf("%d\n", numberOfArithmeticSlices(
        num,
        sizeof(num)/sizeof(num[0])
    ));
    return 0;
}
